datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedPartitionAssignmentStrategy.java

```diff
@@ -233,7 +233,6 @@ private ClusterThroughputInfo fetchPartitionThroughputInfo(DatastreamGroup datas
    */
   @Override
   public void cleanupStrategy() {
-    _assigner.cleanupMetrics();
```

Comment:
You need this call to clean up the registered metrics by LoadBasedPartitionAssigner in cleanupStrategy.

Reply:
Currently there are no metrics registered. Can you explain why I'd need to register/unregister?

Reply:
Without deregistration, the host will continue to emit the metrics until the container is shut down. This will put us in a tricky situation when either the datastream is stopped or datastream task is moved to another host and the host will not stop emitting the metrics. Many times we apply aggregation, which will also repeat incorrect values after transitions.

Reply:
got it. I wondered about that, but I didn't see the other updateOrCreateX() methods do any registration. I'll add back the de/registration, thx.